kolena/dataset/dataset.py

```diff
@@ -182,7 +184,9 @@ def _to_deserialized_dataframe(df: pd.DataFrame, column: str) -> pd.DataFrame:
     )
     flattened = _flatten_composite(flattened)
     flattened = flattened.loc[:, ~flattened.columns.str.endswith(DATA_TYPE_FIELD)]
-    return _dataframe_object_serde(flattened, _deserialize_dataobject)
+    df_post = _dataframe_object_serde(flattened, _deserialize_dataobject)
+    df_post.replace(np.nan, None, inplace=True)
```

Comment:
this is required, b/c `pd.json_normalize` will perform dtypes converting as well.